addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-connection-service.js

```diff
@@ -360,6 +369,108 @@ class EnvironmentScConnectionService extends Service {
     return { password, networkInterfaces: this.toNetworkInterfaces(instanceInfo) };
   }
 
+  async createPrivateSageMakerUrl(requestContext, envId, connection, presign_retries = 10) {
```

Comment:
This method is getting quite large. Should we consider breaking up the code to contain two helper methods.
```
createPrivateSagemakerUrl(...) {
    updateSagemakerPolicy(newPolicy);
    generatePrivateSagemakerUrl();
    updateSagemakerPolicy(originalPolicy);
}
```

Reply:
Yeah, I agree